### DISCUSSION TITLE
Django'da URL için ayarlar bölümü kurmak

### DISCUSSION THREAD

**Discussion by volkantash** [BTK-Django/btk_shop#4] [General]
Selamlar. 

Django uygulamalarındaki url öğelerinin doğru işlemesi için öğretimlerde projenin urls.py dosyası içinde uygulama için taban url dilimi tanıtmamız gerektiği anlatılmakta. 

proje_adı/urls.py
```py
from django.contrib import admin
from django.urls import include, path

urlpatterns = [
    path("admin/", admin.site.urls),
]
```

# Yeni uygulama eklenince
Yeni uygulama eklenince bu dosyanın içini açıp ```path("product/", include("product.urls")),``` satırını yazıyoruz. Ve böylece 127.0.0.1/product ile uygulamayı kullanabiliyoruz.

proje_adı/urls.py
```py
from django.contrib import admin
from django.urls import include, path

urlpatterns = [
    path("product/", include("uygulama_adı.urls")),
    path("admin/", admin.site.urls),
]
```

Ancak adı değiştirmek istediğimizde yine bu dosyanın içine girip elle değiştirmemiz gerekiyor. Oysa bunu yönetici arayüzünden yapmamız gerekir.

# Düşündüğüm çözüm
Projenin url tanıtma dosyasında veritabanından çekilebilecek veriler ile bir çözüm kurabiliriz.

proje_adı/urls.py
```py
from django.contrib import admin
from django.urls import include, path

urlpatterns = [
    path("<slug:uyg_urlsi>/", include("uygulama_adı.urls")),
    path("admin/", admin.site.urls),
]
```
Başka çözümler de çıkacaktır. Buyrun tartışmaya. 

Daha çok bilgi edinmek için [URL dispatcher](https://docs.djangoproject.com/en/4.2/topics/http/urls/#url-dispatcher)

**Comment by volkantash** [comment]
İşlikte (Atölyede) projenin urls.py dosyasında tek bir satırı değiştirerek işimi çözüyorum yanıtı verilmişti. Bugün araştırır iken bunu gördüm. Uygulamadaki urls.py içinde şöyle yazarak istediğimize ulaşabiliyormuşuz. Böylece varmak istediğim yere bir adım daha yaklaştığımı düşünüyorum.

![resim](https://github.com/BTK-Django/btk_shop/assets/25375593/9ebe525f-99e8-4d4b-b5d3-dce957d78d3a)

**Reply by SeherKumsar** [reply]
Django projeleri genellikle birden fazla uygulamayı içerir ve her uygulama, kendi URL yapılandırmasına ve görünüm işlevlerine sahip olabilir. Projedeki ana URL yapılandırması, farklı uygulamalar arasındaki URL'leri düzenlemek ve yönlendirmek için kullanılır. 

Burada belirttiğin örnekte ana yapılandırmada belirli bir uygulamanın sayfalarının URL'lerini yönlendirmek için kullanılıyor.  Bu yönlendirmeler, her uygulamanın views dosyasında tanımlanan fonskiyonlar aracılığıyla çalışır ve ilgili şablonlara(template) yönlendirme yapar. Her bir uygulama kendi içinde birden fazla sayfa bulundurabilir ve bu sayfaların her biri ilgili şablonlara yönlendirilir.

**Reply by volkantash** [reply]
Evet. İşte buradaki yöntemi alıp yönetici arayüzü üzerinden url belirtmek amacı ile bir şeyler yapmak istedim. 

> alanadım . com/product/1  

Burası beni ID'si 1 olan ürüne bağlıyor. İyi hoş. Ama dili Türkçe olan birisine ben böyle url sunmak yerine şu urlyi sunmak istiyorum.

> alanadım . com/ürün/1

Sonra başka diller için de benzer şeyler yapmak istiyorum. Ve bunların hepsini yönetici arayüzünden yapmalıyım.

Bu yukarıdaki örnek ile bunu başarmaya yaklaştım. İlk denemelerimi az önce yaptım. İstediğim oldu. Şimdi biraz daha ileriye taşıyıp yönetici arayüzünde işi daha uzmanca tutmak istiyorum. 

Örneğin AYARLAR diye ayrı uygulama yazıp bütün uygulamaların url değerlerini bu uygulama üzerinden ayarlayabilir miyim sorusuna yanıtlar aramaya çalışıyorum.
